src/pyodide.js

```diff
@@ -2,6 +2,13 @@
  * The main bootstrap script for loading pyodide.
  */
 
+// Regexp for validating package name and URI
+var package_name_regexp = '[a-z0-9_\-]+'
```

Comment:
Also, they *can* include capital letters (but rarely do due to PEP8).

This [SO comment](https://stackoverflow.com/questions/5474008/regular-expression-to-confirm-whether-a-string-is-a-valid-identifier-in-python) may be helpful.  I have no idea if the proposed regex there translates easily into a Javascript regex.

Reply:
Well `python-dateutil` is a package name that is currently used... As far as I understand here package name would differ from the import name, and I think this situation makes sense since multiple packages can have the same import name (e.g. https://github.com/seomoz/simhash-py and https://github.com/leonsim/simhash which both import as `simhash`).

Here the regexp doesn't include capitals, but it is initialized with the [`i` flag](https://github.com/leonsim/simhash) to ignore case.

Reply:
You are right.  Thanks for clearing my confusion.  In this case we need to use the import name, since you can't have two packages with the same import name installed at the same time.  That makes using the filename as a proxy for import name a bit difficult, of course.

I missed the ignore case flag.  That seems fine, then.